Packs/AggregatedScripts/Scripts/IndicatorEnrichment/IndicatorEnrichment.py

```diff
@@ -0,0 +1,999 @@
+import demistomock as demisto
+from CommonServerPython import *
+
+from AggregatedCommandApiModule import *
+
+# =========================
+# Constants & Configuration
+# =========================
+
+MAX_INDICATORS = 100
+
+NO_VALID_INDICATORS_MESSAGE = (
+    "No valid indicators provided. You must provide at least one valid indicator " "in arguments: indicators_list or text."
+)
+
+ERR_INDICATOR_LIMIT_EXCEEDED_TEMPLATE = (
+    "Error: Indicator limit exceeded. Found {found} indicators, but the limit is {limit}. "
+    "To process all indicators, set the ignore_indicator_limit argument to true. "
+    "Note: This should be avoided if possible."
+)
+
+
+class FatalArgException(Exception):
+    """
+    Raised when the command arguments are invalid in a way that must stop the command execution.
+
+    These are surfaced to the user via `return_error` in `main()`.
+    """
+
+
+# Unified configuration for all supported indicator types
+INDICATOR_TYPE_CONFIG: dict[str, dict[str, str]] = {
+    "ip": {
+        "display": "IP",
+        "command": "ip-enrichment",
+        "arg_name": "ip_list",
+        "enrichment_prefix": "IPEnrichment",
+    },
+    "url": {
+        "display": "URL",
+        "command": "url-enrichment",
+        "arg_name": "url_list",
+        "enrichment_prefix": "URLEnrichment",
+    },
+    "domain": {
+        "display": "Domain",
+        "command": "domain-enrichment",
+        "arg_name": "domain_list",
+        "enrichment_prefix": "DomainEnrichment",
+    },
+    "cve": {
+        "display": "CVE",
+        "command": "cve-enrichment",
+        "arg_name": "cve_list",
+        "enrichment_prefix": "CVEEnrichment",
+    },
+    "file": {
+        "display": "File",
+        "command": "file-enrichment",
+        "arg_name": "file_hash",
+        "enrichment_prefix": "FileEnrichment",
+    },
+}
+
+SUPPORTED_INDICATOR_TYPES: set[str] = set(INDICATOR_TYPE_CONFIG.keys())
+
+
+# ===============
+# Helper Functions
+# ===============
+
+
+def normalize_indicator_type(raw_type: str) -> str:
+    """
+    Normalize raw indicator type names into our internal bucket names.
+
+    Examples:
+        - "ip", "IPv4" -> "ip"
+        - "Url" -> "url"
+        - "domain" -> "domain"
+        - "cve" -> "cve"
+        - "md5", "sha1", "sha256" -> "file"
+        - Anything unknown -> returned as-is for unsupported classification.
+
+    Args:
+        raw_type: The raw type string from extractIndicators / auto_detect_indicator_type.
+
+    Returns:
+        Normalized internal type name ("ip", "url", "domain", "cve", "file")
+        or the original string if not recognized (treated as unsupported).
+    """
+    if not raw_type:
+        return "Unknown"
+
+    t = raw_type.lower()
+
+    if t in {"ip", "ipv4"}:
```

Comment:
style choice, i probably would have put this info in a field under INDICATOR_TYPE_CONFIG so there is one source of truth for this definition